debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java

```diff
@@ -124,8 +135,9 @@ private Object convertToLocalDateTime(Column column, String value) {
      * @return the converted value;
      */
     private Object convertToTimestamp(Column column, String value) {
-        if (ALL_ZERO_TIMESTAMP.equals(value) && column.isOptional()) return null;
-        if (ALL_ZERO_TIMESTAMP.equals(value)) value = EPOCH_TIMESTAMP;
+        final boolean matches = ALL_ZERO_TIMESTAMP.matcher(value).matches();
+        if (matches && column.isOptional()) return null;
```

Comment:
Sorry for nit-picking, but let's not use the braceless if-style, I think it's very easy to be mis-read. I'd suggest the following here (and similarly above):

    if (matches) {
        if (column.isOptional()) {
             return null;
        }
        else {
            value = EPOCH_TIMESTAMP;
        }
    }